taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,328 @@
+                                                    9))
+
+        self.assertRaises(ValidationError,
+                          lambda: create_vocabulary(self.repository.id,
+                                                    "vocabulary",
+                                                    "description",
+                                                    "abc",
+                                                    Vocabulary.FREE_TAGGING,
+                                                    9))
+
+        # pass in None, violates constraints
+        self.assertRaises(ValidationError,
+                          lambda: create_vocabulary(self.repository.id,
+                                                    "vocabulary",
+                                                    "description",
+                                                    None,
+                                                    Vocabulary.MANAGED,
+                                                    9))
+
+        self.assertRaises(ValidationError,
+                          lambda: create_vocabulary(self.repository.id,
+                                                    "vocabulary",
+                                                    "description",
+                                                    False,
+                                                    "",
+                                                    9))
+
+    def test_delete_term(self):
+        """
+        Test delete_term
+        """
+        self.assertEquals(1, Term.objects.count())
+        delete_term(self.term.id)
+        self.assertEquals(0, Term.objects.count())
+        self.assertRaises(Http404, lambda: delete_term(9))
+        self.assertRaises(Http404, lambda: delete_term(self.term.id))
+        self.assertEquals(0, Term.objects.count())
+
+    def test_delete_vocabulary(self):
+        """
+        Test delete_vocabulary
+        """
+        self.assertEquals(1, Vocabulary.objects.count())
+        self.assertRaises(Http404, lambda: delete_vocabulary(9))
+        self.assertEquals(1, Vocabulary.objects.count())
+        self.assertRaises(ProtectedError, lambda: delete_vocabulary(
+            self.vocabulary.id))
+        delete_term(self.term.id)
+        delete_vocabulary(self.vocabulary.id)
+        self.assertEquals(0, Vocabulary.objects.count())
+
+    def test_get_learning_resources_for_term(self):
+        """
+        Test get_learning_resources_for_term
+        """
+        self.assertEquals([],
+                          list(get_learning_resources_for_term(self.term.id)))
+
+        self.learning_resource.terms.add(self.term)
+        self.assertEquals([self.learning_resource],
+                          list(get_learning_resources_for_term(self.term.id)))
+
+        self.assertRaises(Http404,
+                          lambda: get_learning_resources_for_term(9))
+
+    def test_get_term(self):
+        """
+        Test get_term
+        """
+        invalid_id = -1
```

Comment:
This fixes one of the test failures. Before, instead of `invalid_id`, this was hard-coded to `9`. During testing, `self.term` was created at least 15 times, so often a term with pk `9` existed.